include/EASTL/intrusive_list.h

```diff
@@ -791,8 +791,8 @@ namespace eastl
 				EASTL_FAIL_MSG("intrusive_list::insert(): element already on a list.");
 		#endif
 
-		node_type& next = *const_cast<node_type*>(pos.mpNode);
-		node_type& prev = *static_cast<node_type*>(next.mpPrev);
+		intrusive_list_node& next = *const_cast<node_type*>(pos.mpNode);
```

Comment:
Yes, this is more of a question about the original code.  I'm not sure the purpose of casting the original type of the data members (which is already intrusive_list_node) to the user-type then back to its base-class.  There is case of removing const but in that case I'm not sure why we are casting to node_type.